internal/store/endpointslice.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2022 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package store
+
+import (
+	"context"
+	"strconv"
+
+	basemetrics "k8s.io/component-base/metrics"
+
+	"k8s.io/kube-state-metrics/v2/pkg/metric"
+	generator "k8s.io/kube-state-metrics/v2/pkg/metric_generator"
+
+	discoveryv1 "k8s.io/api/discovery/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+)
+
+var (
+	descEndpointSliceAnnotationsName     = "kube_endpointslice_annotations"
+	descEndpointSliceAnnotationsHelp     = "Kubernetes annotations converted to Prometheus labels."
+	descEndpointSliceLabelsName          = "kube_endpointslice_labels"
+	descEndpointSliceLabelsHelp          = "Kubernetes labels converted to Prometheus labels."
+	descEndpointSliceLabelsDefaultLabels = []string{"endpointslice"}
+)
+
+func endpointSliceMetricFamilies(allowAnnotationsList, allowLabelsList []string) []generator.FamilyGenerator {
+	return []generator.FamilyGenerator{
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_endpointslice_info",
+			"Information about endpointslice.",
+			metric.Gauge,
+			basemetrics.ALPHA,
+			"",
+			wrapEndpointSliceFunc(func(s *discoveryv1.EndpointSlice) *metric.Family {
+
+				m := metric.Metric{
+					LabelKeys:   []string{"addresstype"},
+					LabelValues: []string{string(s.AddressType)},
+					Value:       1,
+				}
+				return &metric.Family{Metrics: []*metric.Metric{&m}}
+			}),
+		),
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_endpointslice_created",
+			"Unix creation timestamp",
+			metric.Gauge,
+			basemetrics.ALPHA,
+			"",
+			wrapEndpointSliceFunc(func(s *discoveryv1.EndpointSlice) *metric.Family {
+				ms := []*metric.Metric{}
+				if !s.CreationTimestamp.IsZero() {
+					ms = append(ms, &metric.Metric{
+						Value: float64(s.CreationTimestamp.Unix()),
+					})
+				}
+				return &metric.Family{
+					Metrics: ms,
+				}
+			}),
+		),
+		*generator.NewFamilyGeneratorWithStability(
+			"kube_endpointslice_endpoints",
```

Comment:
I think we are plumbing too maybe information inside of this metric. What would you think about moving the conditions, targetref and addresses into their own metrics? There are some precedence to that kind of split with statuses metrics and `kube_endpoint_address`.

Reply:
I decided to combine everything in it that is in https://pkg.go.dev/k8s.io/api/discovery/v1#Endpoint (which is used in 
https://pkg.go.dev/k8s.io/api/discovery/v1#EndpointSlice). This would make it easier for users to query it instead of combining two or more metrics. Do you see any downsides?